deployment/config/public-graphql-api-gateway/gateway.config.ts

```diff
@@ -0,0 +1,44 @@
+// @ts-expect-error not a dependency
+import { createOtlpHttpExporter, defineConfig } from '@graphql-hive/gateway';
+
+const defaultQuery = `#
+# Welcome to the Hive Console GraphQL API.
+#
+`;
+
+export const gatewayConfig = defineConfig({
+  transportEntries: {
+    graphql: {
+      location: process.env['GRAPHQL_SERVICE_ENDPOINT'],
+    },
+  },
+  supergraph: {
+    type: 'hive',
+    endpoint: process.env['SUPERGRAPH_ENDPOINT'],
+    key: process.env['HIVE_CDN_ACCESS_TOKEN'],
+  },
+  graphiql: {
+    title: 'Hive Console - GraphQL API',
+    defaultQuery,
+  },
+  propagateHeaders: {
+    fromClientToSubgraphs({ request }) {
+      return {
+        'x-request-id': request.headers.get('x-request-id'),
```

Comment:
GW should do this automatically actually 😅

Reply:
Request id header is passed to the subgraph http requests automatically.

Reply:
yeah, didn't find it in the docs, but it it forwarding for sure, see https://github.com/graphql-hive/gateway/blob/e888ff1c4755056d5c45ff0b7593b092e1dc909c/packages/runtime/src/plugins/useRequestId.ts#L21 + https://github.com/graphql-hive/gateway/blob/main/packages/runtime/src/createGatewayRuntime.ts#L995